price-estimator/src/main.rs

```diff
@@ -89,9 +99,14 @@ fn main() {
             async move {
                 // TODO: format the response the way the nodejs price estimator did.
                 let orderbook = orderbook.get_reduced_orderbook().await;
-                let result = estimate_price(token_pair, sell_amount_in_quote as f64, orderbook);
+                let result = estimate_buy_amount(
```

Comment:
I thought about this too but wasn't sure what was best. We'll see how it works out when we add the second endpoint. I don't think we should refactor before that.
I like splitting the filter and the handler because of the separation. The filter is responsible for handling the path and extracting the parameters. The handler takes the parameters and computes the result.
With this in mind I don't see anything wrong with touching 3 files. Each file has a purpose.

Reply:
> I like splitting the filter and the handler because of the separation.

I feel like they should be in the same module since they are tightly coupled - changes to one (adding a new parameter for example) usually require changes in the other. I do agree that the internal handler implementation (for example the code that actually calculates the price estimate) should be in its own module.

Reply:
I see what you mean. I will put `fn estimate_buy_amount(...) -> String` into `filter.rs` and keep `fn estimate_buy_amount_internal` in it's own module.